src/TranslationServiceProvider.php

```diff
@@ -1,38 +1,39 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Spatie\TranslationLoader;
 
 use Illuminate\Support\Str;
-use Illuminate\Translation\FileLoader;
 use Illuminate\Translation\TranslationServiceProvider as IlluminateTranslationServiceProvider;
 
 class TranslationServiceProvider extends IlluminateTranslationServiceProvider
 {
     /**
      * Register the application services.
      */
-    public function register()
+    public function register(): void
```

Comment:
You could replace a lot of this logic by using `spatie/laravel-package-tools`. Take a look at spatie/laravel-health for a good example

Reply:
I'm not sure how to do that because for this I must extend the `Spatie\LaravelPackageTools\PackageServiceProvider` but the current ServiceProvider already extends `Illuminate\Translation\TranslationServiceProvider`.

Reply:
Okay, I managed to update the logic using the `spatie/laravel-package-tools`, but I'm not sure my solution is good enough:

```php
<?php

declare(strict_types=1);

namespace Spatie\TranslationLoader;

use Illuminate\Translation\TranslationServiceProvider as IlluminateTranslationServiceProvider;
use Spatie\LaravelPackageTools\Package;
use Spatie\LaravelPackageTools\PackageServiceProvider;

class TranslationServiceProvider extends PackageServiceProvider
{
    public function configurePackage(Package $package): void
    {
        $package
            ->name('laravel-translation-loader')
            ->hasConfigFile()
            ->publishesServiceProvider(TranslationServiceProvider::class)
            ->hasMigrations('create_language_lines_table');
    }

    /**
     * Register the application services.
     */
    public function register(): void
    {
        parent::register();

        (new IlluminateTranslationServiceProvider($this->app))->register();

        $this->registerLoader();
    }

    /**
     * Register the translation line loader. This method registers a
     * `TranslationLoaderManager` instead of a simple `FileLoader` as the
     * applications `translation.loader` instance.
     */
    protected function registerLoader(): void
    {
        $this->app->singleton('translation.loader', function ($app) {
            $class = config('translation-loader.translation_manager');

            return new $class($app['files'], $app['path.lang']);
        });
    }
}
```

If I don't do `(new IlluminateTranslationServiceProvider($this->app))->register();` there's this error:
```
Target class [translator] does not exist.
```

I can fix the error in tests by ensuring the `IIlluminate\Translation\TranslationServiceProvider` is loaded but the problem persists in a real Laravel app, so I've added it in the `register()` method here. This fixes the issue, the tests are passing, and testing in a real Laravel app succeeds, but I don't know if this is a good practice.

Any suggestions?

Reply:
Fixed it.